google/cloud/storage/emulator/gcs/holder.py

```diff
@@ -59,21 +59,31 @@ def init_upload(
 
     @classmethod
     def init_resumable_rest(cls, request, bucket):
-        name = request.args.get("name", "")
+        name = request.args.get("name", None)
         rest_only = {}
-        if len(request.data) > 0 and request.data.decode("utf-8") != "{}":
-            if name != "":
-                utils.error.invalid("name argument in non-empty payload", None)
+        if len(request.data) > 0:
             data = json.loads(request.data)
             rest_only = cls.__extract_rest_only(data)
             metadata = json_format.ParseDict(data, resources_pb2.Object())
+            if name != None and name != metadata.name:
```

Comment:
Question: when should one use `is not None` vs. `!= None` ?

Reply:
`is not` - compares object addresses
`!=` - compares object values `__eq__(self, obj)`  or literal values

Python uses [String Interning](https://en.wikipedia.org/wiki/String_interning), but it's an implementation detail IIUC so I decided to use `!=` to compare string literal values and `is not` for `None`.

I honestly took this for granted, and found a helpful article that walked me through it: https://realpython.com/python-is-identity-vs-equality/